ambry-store/src/main/java/com.github.ambry.store/BlobStoreCompactor.java

```diff
@@ -123,6 +125,7 @@ public boolean accept(File dir, String name) {
     this.time = time;
     this.sessionId = sessionId;
     this.incarnationId = incarnationId;
+    this.useDirectIO = Platform.isLinux() && config.storeCompactionEnableDirectIO;
```

Comment:
`Platform.isLinux()` is implemented by `System.getProperty("os.name").startsWith("Linux")`

I checked some other library, for example, `[commons.apache.org.SystemUtils](https://commons.apache.org/proper/commons-lang/apidocs/src-html/org/apache/commons/lang3/SystemUtils.html#line.1066)` checks both `Linux` and `LINUX`.
`toLowerCase()` might be a safer solution, in case some distributions use 'LINUX'.